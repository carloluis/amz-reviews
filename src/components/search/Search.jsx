import React from 'react';
import PropTypes from 'prop-types';
import styles from './Search.scss';

class Search extends React.PureComponent {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }
    handleChange(event) {
        this.props.onUpdateSearch(event.target.value);
    }
    render() {
        return (
            <div className={styles.container}>
                <input
                    className={styles.search}
                    onChange={this.handleChange}
                    value={this.props.value}
                    placeholder="Search"
                    type="text"
                />
            </div>
        );
    }
}

Search.propTypes = {
    onUpdateSearch: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
};

export default Search;
