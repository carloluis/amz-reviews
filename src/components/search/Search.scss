@import '../../styles/palette.scss';

.container {
    display: inline-block;
    position: relative;
    width: 100%;

    &::after {
        content: 'âš²';
        display: inline-block;
        transform: rotate(-45deg);
        transform-origin: center;
        font-size: 1.4rem;
        font-weight: bold;
        color: $blue;
        position: absolute;
        right: 16px;
    }

    .search {
        position: relative;
        border: 1px solid $blue;
        border-radius: 16px;
        width: inherit;
        height: 32px;
        padding: 4px 32px 4px 24px;
        font-size: 14px;
        color: $blue;

        &:focus {
            outline-offset: 0;
            outline: none;
        }

        &::placeholder {
            color: $blue;
        }
    }
}
