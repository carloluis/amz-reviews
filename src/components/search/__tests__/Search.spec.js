import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Search from '../Search';

const PROPS = {
    onUpdateSearch: jest.fn(),
    value: 'some'
};

describe('<Search/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<Search {...PROPS} />).toJSON();
        expect(result).toMatchSnapshot();
    });
});

describe('Search', () => {
    it('should call the onUpdateSearch prop when change input value', () => {
        const wrapper = shallow(<Search {...PROPS} />);
        const event = { target: { value: 'x' } };
        wrapper.find('input.search').simulate('change', event);
        expect(PROPS.onUpdateSearch).toHaveBeenCalled();
    });
});
