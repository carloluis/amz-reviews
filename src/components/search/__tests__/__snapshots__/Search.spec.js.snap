// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> snapshot 1`] = `
<div
  className="container"
>
  <input
    className="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="some"
  />
</div>
`;
